### ISSUE TITLE
Put git branch name in message?

### ISSUE THREAD

**Issue by tecnobrat** [jlewallen/jenkins-hipchat-plugin#12]
Any way to put the git branch name into the message sent to hipchat?

**Comment by rbroemeling** [comment]
I just wanted to add in that we would also find this advantageous.  Current messages look like:

[project] - #7 Success after 51 min (Open)

... it would be great if they could be changed to look like:

[project]/[branch] - #7 Success after 51 min (Open)

I checked out the plugin code and the changes look fairly straightforward, however I couldn't figure out where to get the current branch name from the Hudson/Jenkins API, so that has put me at an impasse.

**Comment by rocketraman** [comment]
+1

**Comment by jmcclell** [comment]
It would be nice to be able to configure messages all-together, actually. Maybe some kind of message templating.

**Comment by Mange** [comment]
:+1:

**Comment by kmullin** [comment]
:+1: as well.

**Comment by ghost** [comment]
I also think this is a good thing.

**Comment by kmullin** [comment]
Ok, so I took a stab at this. I found two approaches:
1. Look at ChangeLogSet to see if entries, read commitId(Msg,Author) from last entry in changes
   changes: https://github.com/uservoice/jenkins-hipchat-plugin/compare/add_commit_hash
2. HACK: basically grep build log for specific string to match branch and SHA1 hash
   heavily relies on a very short buildLog (< 100 lines)
   also there are specific hard codings for us in there
   changes: https://github.com/uservoice/jenkins-hipchat-plugin/compare/uservoice_hacks

I also looked into trying to use [hudson.scm.SCM](http://javadoc.jenkins-ci.org/hudson/scm/SCM.html), but I couldn't figure out how to expose SCM related info... maybe someone more knowledgeable could help solve that.

These changes were branched from [hipchat-0.1.4 release](https://github.com/jlewallen/jenkins-hipchat-plugin/commit/6cb438d9203c685e0f7feea609aa76fc6892f338) because it was the only known working version that we'd already been using.

**Comment by kscaldef** [comment]
Yes, message templating similar to email notifications would be really useful.  For our parameterized builds, it would be much more useful to be able to include some of the parameters in the notification.

**Comment by colszowka** [comment]
:heavy_plus_sign:  :one:

**Comment by marcins** [comment]
Vote for having a per-project message template too. My use case is that I have a project with a parameter that determines whether to deploy to staging or production and in HipChat there's no way to tell which kind of build was run.

**Comment by marcins** [comment]
If anyone wants to try the plugin with Mustache notification templates I have a snapshot HPI up in Dropbox:

https://dl.dropbox.com/u/5336861/hipchat.hpi

YMMV, disclaimer, etc, etc :-)  

Also check the [updated README](https://github.com/marcins/jenkins-hipchat-plugin/tree/feature-mustache-messages)

**Comment by marcins** [comment]
FYI - It appears that the git plugin sets an environment variable GIT_BRANCH based on which branch is being built, so with the Mustache template you can output `{{build.environment.GIT_BRANCH}}`.

**Comment by borkdude** [comment]
+1

**Comment by JustinAiken** [comment]
:+1:

**Comment by tokyowizard** [comment]
+1

**Comment by dennisbaskin** [comment]
+1
